components/03-modules/slider/slider.js

```diff
@@ -5,104 +5,125 @@
 
   const sliders = [...document.querySelectorAll('.slider')];
 
-  sliders.forEach(slider => {
-    const dataValues   = slider.dataset,
-          items        = slider.querySelectorAll('.slider__item').length,
-          dotContainer = slider.querySelector('.slider__dot-navigation');
-
-    let config = {
-      classNameFrame: 'slider__handler',
-      classNameSlideContainer: 'slider__slides',
-      classNamePrevCtrl: 'slider__prev',
-      classNameNextCtrl: 'slider__next',
-      slidesToScroll: 1
-    };
-
-    Object.keys(dataValues).map((objectKey) => {
-      if (objectKey != 'slidesToScroll') {
-        if (!isNaN(parseInt(dataValues[objectKey]))) {
-          config[objectKey] = parseInt(dataValues[objectKey]);
+  class Slider {
+    constructor(slider, dataValues) {
+      this.active = 'slider__dot--active';
+      this.config = {
+        classNameFrame         : 'slider__handler',
+        classNameSlideContainer: 'slider__slides',
+        classNamePrevCtrl      : 'slider__prev',
+        classNameNextCtrl      : 'slider__next',
+        slidesToScroll         : 1
+      };
+      this.dataValues = dataValues;
+      this.slider = slider;
+      this.dotContainer = this.slider.querySelector('.slider__dot-navigation');
+      this.items = this.slider.querySelectorAll('.slider__item').length;
+      this.navigationContainer = this.slider.querySelector('.slider__navigation');
+      this.dotCount = 0;
+
+      if (this.dotContainer) {
+        this.dots = this.dotContainer.childNodes;
+
+        this.slider.addEventListener('before.lory.init', () => {
+          this.cloneDots()
+          this.addDotListeners();
+        });
+
+        this.slider.addEventListener('after.lory.slide', (event) => {
+          this.dotClassToggle(event);
+        });
+
+        this.slider.addEventListener('on.lory.resize', () => {
+          this.lorySlider.slideTo(0);
+          this.dotContainer.innerHTML = '';
+
+          this.cloneDots();
+          this.addDotListeners();
+
+          this.dotContainer.childNodes[0].classList.add(this.dotActive);
+        });
+      }
+
+      Object.keys(this.dataValues).map((objectKey) => {
+        if (objectKey !== 'slidesToScroll') {
+          if (!isNaN(parseInt(this.dataValues[objectKey]))) {
+            this.config[objectKey] = parseInt(this.dataValues[objectKey]);
+          }
+          else {
+            this.config[objectKey] = this.dataValues[objectKey];
+          }
         }
-        else {
-          config[objectKey] = dataValues[objectKey];
+      });
+
+      this.lorySlider = lory(this.slider, this.config);
+    }
+
+    cloneDots() {
```

Comment:
Something is wrong with dots, b/c on small screen you've 6 dots, and on larger 4 dots.

Reply:
As we talked on chat. It's ok.